core/src/main/java/com/linecorp/armeria/common/util/Unwrappable.java

```diff
@@ -87,4 +87,11 @@ default <T> T as(Class<T> type) {
     default Unwrappable unwrap() {
         return this;
     }
+
+    /**
+     * TBU.
+     */
+    default <U extends Unwrappable> U root() {
```

Comment:
Note that this is an unsafe cast since we don't know the concrete type of this object.

I think the options we have are..

1. Just return `Unwrappable` and let the user cast if they want to
2. It's unclear what behavior we really want. Only implement `root` for `RequestContext` with minimal constraints for now, and generalize later if necessary.
3. Unsafe cast as in this PR

Reply:
Modified to just return `Unwrappable` and let users cast if they want.